packages/connect-webextension/src/proxy/index.ts

```diff
@@ -0,0 +1,134 @@
+import EventEmitter from 'events';
+
+// NOTE: @trezor/connect part is intentionally not imported from the index due to NormalReplacementPlugin
+// in packages/suite-build/configs/web.webpack.config.ts
+import {
+    ERRORS,
+    IFRAME,
+    POPUP,
+    createErrorMessage,
+    ConnectSettings,
+    Manifest,
+    CallMethod,
+} from '@trezor/connect/lib/exports';
+import { factory } from '@trezor/connect/lib/factory';
+
+import { parseConnectSettings } from '../connectSettings';
+import { WindowServiceWorkerChannel } from '../channels/window-serviceworker';
+
+const eventEmitter = new EventEmitter();
+let _settings = parseConnectSettings();
+let _channel: any;
+
+const manifest = (data: Manifest) => {
+    _settings = parseConnectSettings({
+        ..._settings,
+        manifest: data,
+    });
+};
+
+const dispose = () => {
+    eventEmitter.removeAllListeners();
+    _settings = parseConnectSettings();
+
+    return Promise.resolve(undefined);
+};
+
+const cancel = () => {
+    if (_channel) {
+        _channel.clear();
+    }
+};
+
+const init = (settings: Partial<ConnectSettings> = {}): Promise<void> => {
```

Comment:
ok, so expected way to use this is that in the foreground, I need to call `TrezorConnect.init` exactly once. right? is that developer friendly actually?
Wouldn't it be better to force users to only do `init` inside service worker (which could also init channel, etc) and leave this from proxy entirely? (throw error "do this in service worker" instead?)

Reply:
The idea I had in mind when doing it like that  that TrezorConnect is initialize wherever the developer is actually using TrezorConnect.
In the example that this PR contains using connect-explorer TrezorConnect is used in the foreground and initialized there and that is handy for us because we can use the `settings` page in connect-explorer to initialize TrezorConnect with different `connectSrc` and `trustedHost`. 

We could do it in a way that both options are possible. 

If TrezorConnect is initalized in Service Worker then it is not necessary to initialize it in foreground, but if you initialize it in foreground and the settings are different that the one initially it initializes it again. WDYT?

Reply:
ok,
1. having an option to initialize from frontend makes sense. lets keep it
2. just lets make sure that we don't have some unnecessary logic in frontend.